help/en/releasenotes/current-draft-warnings.shtml

```diff
@@ -1,6 +1,10 @@
 
     <!-- contains new warnings for the release under development -->
-    <li>
+    <li>DecoderPro has been added to the permission system. Due to the
+        complexity of many different programmers on many different
+        hardware, there might be bugs that we haven't noticed yet. If
+        something isn't working as expected, please report it on the
+        <a href="https://groups.io/g/jmriusers/topics">JMRI users group.</a>
```

Comment:
Note to reviewers:
`jmri.jmrix.can.cbus.swing.nodeconfig.NodeConfigToolPane` does a type casting to `CbusDccProgrammer` which after my changes caused a ClassCastException. But this exception didn't caused a failure in the test suite, despite happening when running the tests. I think the reason for that is that this code is run from SwingUtilities.invokeAndWait(). I think I have solved this particular problem, but I don't have any hardware to test it.

The point is that this PR makes ProgrammerManager to return a PermissionProgrammer instead of a connection specific programmer. And if connection specific code tries to cast the programmer to the connection specific programmer, an exception will occur. And that exception might not be discovered by the test suite.